### PR TITLE
Water shader improvements

### PR DESCRIPTION
**Pull request by BlackYps**
current shader:
old

new shader:
new

The main visual difference is that the overtuned sky reflections for low angles are now more sensible and the weird interpolation near the shores is now gone. This could also help with properly identifying the shore line on some maps.

On a technical level the shader now ignores the values of unitreflectionAmount and skyreflectionAmount that can be set by the map.
I tested it on some maps, but more testing would be good to make sure this doesn't produce unexpected results. I mainly tested on Seton's Clutch, Roanoke Abyss, Saltrock Colony 8 Players, Enfilade and Dual Gap Reloaded.

### CODE REVIEW

**Comment by Garanas** [comment]
Looks a lot better to me, I'll test it on some of my own maps.

**Review by Garanas — APPROVED** [review]

**Comment by Garanas** [comment]
Experimented a bit, water looks a lot more consistent. I can take the loss of 'proper' lava, as that is not used at the moment anyway.

File: effects/water2.fx [hunk 1]
```diff
@@ -378,8 +378,11 @@ float4 HighFidelityPS( VS_OUTPUT inV,
    	//Schlick approximation for fresnel
     float NDotV = saturate(dot(viewVector, N));
 	float F0 = 0.08;
-    float fresnel = F0 + (1.0 - F0) * pow(1.0 - NDotV, 5.0);         
-    refractedPixels = lerp(refractedPixels, reflectedPixels, fresnel);
+    float fresnel = F0 + (1.0 - F0) * pow(1.0 - NDotV, 5.0);
+
+	if (skyreflectionAmount == 1.5)
```
Comment (Garanas) [inline effects/water2.fx:383]: Please add a comment here, stating why it is there

Comment (Garanas) [reply effects/water2.fx:383]: Why do we want this exception here exactly?

**Comment by Garanas** [comment]
It happened again 😞 , I thought I switched the branch. Apologies 😄

**Comment by Garanas** [comment]
Back to just being about water

**Comment by Garanas** [comment]
I tried to force push the fix, but it didn't affect this branch. Not sure if I force pushed your local branch on accident.

File: effects/water2.fx (hunk 1 above)
Comment (BlackYps) [reply effects/water2.fx:384]: If the map author didn't change the skyreflectionAmount, it defaults to 1.5 which is way too bright. With this we can change the default while still preserving the value that the map author deliberately set. I tested it on various maps and there are some maps that set it to even less than one and it does look nicer to preserve that most of the time.

**Pull request merged**